Datadog Dashboard creation failed with plugin crash

### Terraform Version

V0.13

### Affected Resource(s)

-   datadog_dashboard


### Terraform Configuration Files

Sample DD dashboard output from Datadog-to-Terraform Converter

### Panic Output

Stack trace from the terraform-provider-datadog_v3.4.0 plugin:

panic: interface conversion: interface {} is nil, not map[string]interface {}

goroutine 46 [running]:
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogQueryValueRequests(0xc000960c88, 0xc000c7e270)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:3565 +0x1133
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogQueryValueDefinition(0xc000c7e270, 0xc000c7e240)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:3463 +0xe5
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidget(0xc000c7e240, 0xc0002ae800, 0x0, 0x0)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:952 +0x1085
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidgets(0xc000960fb0, 0xc000c7e4e0, 0x168fb0e, 0x6)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:869 +0xee
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogGroupDefinition(0xc000c7e4e0, 0xc0011ffbf0, 0x169afb5, 0x10)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:1231 +0x513
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidget(0xc0011ffbf0, 0xc0002ae400, 0x0, 0x0)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:884 +0x345
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogWidgets(0xc0009612c0, 0x168fb0e, 0x6, 0x13caba0)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:869 +0xee
github.com/terraform-providers/terraform-provider-datadog/datadog.buildDatadogDashboard(0xc000d7fa80, 0xc0007ee270, 0x0, 0xc000d98830)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:351 +0x331
github.com/terraform-providers/terraform-provider-datadog/datadog.resourceDatadogDashboardCreate(0x18e4818, 0xc00035f620, 0xc000d7fa80, 0x1396d60, 0xc00062da80, 0xc0007ee260, 0xaa61ea, 0xc00087c5c0)
	github.com/terraform-providers/terraform-provider-datadog/datadog/resource_datadog_dashboard.go:140 +0xa5
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0xc0003aa9a0, 0x18e47a8, 0xc001112300, 0xc000d7fa80, 0x1396d60, 0xc00062da80, 0x0, 0x0, 0x0)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:330 +0x17f
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc0003aa9a0, 0x18e47a8, 0xc001112300, 0xc00101c150, 0xc00087c5c0, 0x1396d60, 0xc00062da80, 0x0, 0x0, 0x0, ...)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/resource.go:456 +0x67b
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc00000c030, 0x18e47a8, 0xc001112300, 0xc00067a4b0, 0xc001112300, 0x15e6c20, 0xc001046200)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.7.0/helper/schema/grpc_provider.go:955 +0x8ef
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc000093240, 0x18e4850, 0xc001112300, 0xc001238000, 0xc000093240, 0xc0010462d0, 0xc000343ba0)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/server/server.go:332 +0xb5
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler(0x15e6c20, 0xc000093240, 0x18e4850, 0xc0010462d0, 0xc000114540, 0x0, 0x18e4850, 0xc0010462d0, 0xc000ea4000, 0x12273)
	github.com/hashicorp/terraform-plugin-go@v0.3.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0003c6380, 0x18edb78, 0xc000103680, 0xc00106a000, 0xc0003f91a0, 0x2241c40, 0x0, 0x0, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0x52b
google.golang.org/grpc.(*Server).handleStream(0xc0003c6380, 0x18edb78, 0xc000103680, 0xc00106a000, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc000120270, 0xc0003c6380, 0x18edb78, 0xc000103680, 0xc00106a000)
	google.golang.org/grpc@v1.32.0/server.go:859 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x1fd

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
╷
│ Error: Plugin did not respond
│ 
│   with module.datadog_dashboard.datadog_dashboard.Sample-STG,
│   on ../../modules/datadogdashboards/Sample/main.tf line 1, in resource "datadog_dashboard" "Sample-STG":
│    1: resource "datadog_dashboard" "Sample-STG" {

### Expected Behavior

Datadog dashboard created with terraform apply after successful terraform plan

### Actual Behavior

Terraform plan was successful followed by plugin crash with terraform apply

### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:

1. `terraform plan`
2. terraform apply

--- a/datadog/resource_datadog_dashboard.go
+++ b/datadog/resource_datadog_dashboard.go
@@ -1601,6 +1601,9 @@ func getChangeRequestSchema() map[string]*schema.Schema {
 func buildDatadogChangeRequests(terraformRequests *[]interface{}) *[]datadogV1.ChangeWidgetRequest {
 	datadogRequests := make([]datadogV1.ChangeWidgetRequest, len(*terraformRequests))
 	for i, request := range *terraformRequests {
+		if request == nil {
+			continue
+		}
 		terraformRequest := request.(map[string]interface{})
 		// Build ChangeRequest
 		datadogChangeRequest := datadogV1.NewChangeWidgetRequest()
@@ -1820,6 +1823,9 @@ func getDistributionRequestSchema() map[string]*schema.Schema {
 func buildDatadogDistributionRequests(terraformRequests *[]interface{}) *[]datadogV1.DistributionWidgetRequest {
 	datadogRequests := make([]datadogV1.DistributionWidgetRequest, len(*terraformRequests))
 	for i, r := range *terraformRequests {
+		if r == nil {
+			continue
+		}
 		terraformRequest := r.(map[string]interface{})
 		// Build DistributionRequest
 		datadogDistributionRequest := datadogV1.NewDistributionWidgetRequest()
@@ -2433,6 +2439,9 @@ func getHeatmapRequestSchema() map[string]*schema.Schema {
 func buildDatadogHeatmapRequests(terraformRequests *[]interface{}) *[]datadogV1.HeatMapWidgetRequest {
 	datadogRequests := make([]datadogV1.HeatMapWidgetRequest, len(*terraformRequests))
 	for i, r := range *terraformRequests {
+		if r == nil {
+			continue
+		}
 		terraformRequest := r.(map[string]interface{})
 		// Build HeatmapRequest
 		datadogHeatmapRequest := datadogV1.NewHeatMapWidgetRequest()
@@ -3562,6 +3571,9 @@ func getQueryValueRequestSchema() map[string]*schema.Schema {
 func buildDatadogQueryValueRequests(terraformRequests *[]interface{}) *[]datadogV1.QueryValueWidgetRequest {
 	datadogRequests := make([]datadogV1.QueryValueWidgetRequest, len(*terraformRequests))
 	for i, r := range *terraformRequests {
+		if r == nil {
+			continue
+		}
 		terraformRequest := r.(map[string]interface{})
 		// Build QueryValueRequest
 		datadogQueryValueRequest := datadogV1.NewQueryValueWidgetRequest()
@@ -3829,6 +3841,9 @@ func getQueryTableRequestSchema() map[string]*schema.Schema {
 func buildDatadogQueryTableRequests(terraformRequests *[]interface{}) *[]datadogV1.TableWidgetRequest {
 	datadogRequests := make([]datadogV1.TableWidgetRequest, len(*terraformRequests))
 	for i, r := range *terraformRequests {
+		if r == nil {
+			continue
+		}
 		terraformRequest := r.(map[string]interface{})
 		// Build QueryTableRequest
 		datadogQueryTableRequest := datadogV1.NewTableWidgetRequest()
@@ -4606,6 +4621,9 @@ func buildDatadogGeomapDefinition(terraformDefinition map[string]interface{}) *d
 func buildDatadogGeomapRequests(terraformRequests *[]interface{}) *[]datadogV1.GeomapWidgetRequest {
 	datadogRequests := make([]datadogV1.GeomapWidgetRequest, len(*terraformRequests))
 	for i, r := range *terraformRequests {
+		if r == nil {
+			continue
+		}
 		terraformRequest := r.(map[string]interface{})
 		// Build Geomap Request
 		datadogGeomapRequest := datadogV1.NewGeomapWidgetRequest()
@@ -5572,6 +5590,9 @@ func buildDatadogFormulaAndFunctionProcessQuery(data map[string]interface{}) dat
 func buildDatadogTimeseriesRequests(terraformRequests *[]interface{}) *[]datadogV1.TimeseriesWidgetRequest {
 	datadogRequests := make([]datadogV1.TimeseriesWidgetRequest, len(*terraformRequests))
 	for i, r := range *terraformRequests {
+		if r == nil {
+			continue
+		}
 		terraformRequest := r.(map[string]interface{})
 		// Build TimeseriesRequest
 		datadogTimeseriesRequest := datadogV1.NewTimeseriesWidgetRequest()
@@ -5852,6 +5873,9 @@ func getToplistRequestSchema() map[string]*schema.Schema {
 func buildDatadogToplistRequests(terraformRequests *[]interface{}) *[]datadogV1.ToplistWidgetRequest {
 	datadogRequests := make([]datadogV1.ToplistWidgetRequest, len(*terraformRequests))
 	for i, r := range *terraformRequests {
+		if r == nil {
+			continue
+		}
 		terraformRequest := r.(map[string]interface{})
 		// Build ToplistRequest
 		datadogToplistRequest := datadogV1.NewToplistWidgetRequest()

